### PR TITLE
Separate message package

### PR DESCRIPTION
**Pull request by mhubii**
Hi @ARoefer , in order to facilitate ROS 2 (refer, #2) and minimal code changes, it would be necessary to separate out the message package.

It has been a while since I last coded ROS 1 so wasn't able to check thoroughly. Could you please give me a helping hand here? Thank you so much.

### CODE REVIEW

**Comment by raabuchanan** [comment]
Hi @mhubii thank you for your contribution. We're now interested in having a ROS2 wrapping so I'll be taking a look at your PR and will get back to you.

**Comment by mhubii** [comment]
that's cool, moved it here in the meantime https://github.com/ros-ai/ros2_sam

**Comment by mhubii** [comment]
if I recall correctly, this PR separates the message package for ROS 1 so ROS 1/2 branches can be maintained simultaneously as per @ARoefer 's request. But then somehow got lost

File: ros_sam_msgs/CMakeLists.txt
Comment (mhubii) [inline ros_sam_msgs/CMakeLists.txt:1]: creates a separate message packager `ros_sam_msgs`

**Comment by mhubii** [comment]
porting your rqt plugin https://github.com/ipab-slmc/rqt_image_view_seg would be the missing bit here

**Comment by ARoefer** [comment]
Hi @mhubii, I was still aware of it, but I just never found the time to get my hands on a ROS2 and review this. Luckily @raabuchanan now has the opportunity to do so. Sorry about the negligence.

**Comment by mhubii** [comment]
no worries, so this PR here is ROS 1. It simply separates the message package because ROS 2 (as far as I am ware of) cannot have message definitions within the package itself

**Comment by mhubii** [comment]
tried to reach you on linkedin in this regard but couldn't get through :/

**Comment by ARoefer** [comment]
Yeah, I almost never check my LinkedIn :/

**Comment by mhubii** [comment]
I am more than happy to revive this as shared infrastructure is to the benefit of us all, but there isn't much I can do really without reviews.

My PRs targeted ROS 1, and ROS 2, and were vastly left unanswered, just like my message on linkedin. Hope it is somewhat relatable that I moved the ROS 2 implementation elsewhere, as I am in need of this infrastructure and simply couldn't continue waiting. I feel like I have gone out of my way to contribute here.

**Comment by ARoefer** [comment]
You are right, and there is no excuse for that. I reviewed your changes and am not a fan of the way that a ros package contains another ros package. Also it seems that the CMakeLists of the original `ros_sam` moves into `ros_sam_msgs` while `ros_sam` gets a new CMakeLists. That's a bit odd. I have made your changes (minus black-formatting) here: https://github.com/robot-learning-freiburg/ros_sam/pull/10. I would merge those, if they work for you (if you have the time and patience to review them).

**Comment by mhubii** [comment]
I mean this is your repo, but usually not a fan of hiding contributions. Your PR #10 implements the same concept but will remove any of my contributions from the log.  

There is no ros package inside a ros package and this is a common pattern, also in ros 1, you may e.g. refer to bigger bundles like https://github.com/ros-controls/ros_control or what not.

Please do keep in mind you can drop compatibility with noetic eol if this pattern is something you dislike.

**Comment by mhubii** [comment]
the way this could work is

- you review and request changes, I add these changes
- you create a dev branch and merge into dev, then make adjustments yourself 
- I create a new PR

but in the face of progress on this I am happy to hide I contributed here

**Comment by ARoefer** [comment]
I'm sorry, it was not my intent to hide your contribution, I just wanted to save you from having to commit any additional time into this. I have created a new `devel` branch, if you want to redirect your PR there, I can make the changes I would like. If you're willing to spend more time on it, it would be nice if you could add the directory structure present in https://github.com/robot-learning-freiburg/ros_sam/tree/issue/9-separate-message-package. Might be that that kind of substructuring might exist, but I have only ever come across side-by-side packages, even in bigger repos.

If you'd prefer a full review, we can also do that. Whatever you would prefer. Really don't mean to sideline you.

**Comment by mhubii** [comment]
what formatter are you using

**Comment by ARoefer** [comment]
None. I don't like any of them.

**Comment by mhubii** [comment]
hm okay.

I see your point, should rather create a new package and modify the old vs what I did here. I currently don't have the means to make the adjustments. Feel free to close this.

**Comment by ARoefer** [comment]
Okay. I am sorry about this entire process and thank you for trying. As a solace, I hope that the ROS2 version will work out better so that you get the credit you deserve.

Cheers!

**Comment by mhubii** [comment]
okay, I left you some reviews, but believe this was merged

**Comment by mhubii** [comment]
it's a shame really

**Comment by mhubii** [comment]
this is really not how PRs should be treated @raabuchanan, but I hope you can see this yourself

**Comment by raabuchanan** [comment]
@mhubii I'm very sorry you feel mistreated. I assure you that this has been a case of miscommunication with no malice intended. We very much appreciate your contribution to this project and I personally apologize for taking a long time to respond.

**Pull request closed without merging**